test/hotspot/jtreg/containers/cgroup/NestedCgroup.java

```diff
@@ -0,0 +1,222 @@
+ * @library /testlibrary /test/lib
+ * @run main/othervm NestedCgroup
+ */
+
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import jdk.internal.platform.Metrics;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.Platform;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jtreg.SkippedException;
+
+public class NestedCgroup {
+    private static abstract class Test {
+        public static final String CGROUP_OUTER = "jdktest" + ProcessHandle.current().pid();
+        public static final String CGROUP_INNER = "inner";
+
+        // A real usage on x86_64 fits in 39 MiB.
+        public static final int MEMORY_MAX_OUTER = 500 * 1024 * 1024;
+        public static final int MEMORY_MAX_INNER = MEMORY_MAX_OUTER * 2;
+
+        class HookResult {
+            int integerLimit;
+            OutputAnalyzer output;
+        };
+
+        public static String memory_max_filename;
+        public static boolean isCgroup2;
+        public static String controller;
+
+        public static void args_add_cgexec(List<String> args) {
+            args.add("cgexec");
+            args.add("-g");
+            args.add(controller + ":" + CGROUP_OUTER + "/" + CGROUP_INNER);
+        }
+
+        public static String jdkTool;
+
+        public static void args_add_self(List<String> args) {
+            args.add(jdkTool);
+            args.add("-cp");
+            args.add(System.getProperty("test.classes"));
+        }
+
+        public static void args_add_self_verbose(List<String> args) {
+            args_add_self(args);
+            args.add("-XshowSettings:system");
+            args.add("-Xlog:os+container=trace");
+        }
+
+        public Test(String controller_) throws Exception {
+            controller = controller_;
+            OutputAnalyzer output;
+
+            List<String> cgdelete = new ArrayList<>();
+            cgdelete.add("cgdelete");
+            cgdelete.add("-r");
+            cgdelete.add("-g");
+            cgdelete.add(controller + ":" + CGROUP_OUTER);
+            output = ProcessTools.executeProcess(new ProcessBuilder(cgdelete));
+            if (output.contains("cgdelete: libcgroup initialization failed: Cgroup is not mounted")) {
+                throw new SkippedException("cgroup/cgroup2 is not mounted: " + output.getStderr());
+            }
+            output.stdoutShouldBeEmpty();
+            if (!output.contains("cgdelete: cannot remove group '" + CGROUP_OUTER + "': No such file or directory")) {
+                output.stderrShouldBeEmpty();
+            }
+
+            // Alpine Linux 3.20.1 needs cgcreate1 otherwise:
+            // cgcreate: can't create cgroup [...]: No such file or directory
```

Comment:
_"Create the parent cgroup hierarchy"_ would be commenting an obvious thing so such comment is best to rather remove. The problem my comment is trying to describe is that on normal (tested Fedora) systems it is OK to do only:
```
cgcreate -g memory:a/b
```
On Alpine Linux it does not work (I did not investigate why). On Alpine Linux one has to do:
```
cgcreate -g memory:a
cgcreate -g memory:a/b
```
Developers probably will not use Alpine Linux so they may delete the first line
```
cgcreate -g memory:a
```
as it looks redundant. But it will break Alpine Linux.

```suggestion
            // Alpine Linux 3.20.1 needs cgcreate1 as while regular Linuxes are fine with:
            //   cgcreate -g memory:a/b
            // Alpine Linux 3.20.1 needs both commands:
            //   cgcreate -g memory:a
            //   cgcreate -g memory:a/b
            // With only the second line Alpine Linux 3.20.1 would error with:
            //   cgcreate: can't create cgroup [...]: No such file or directory
```